activemodel/test/cases/validations/length_validation_test.rb

```diff
@@ -456,4 +456,22 @@ def test_validates_length_of_using_proc_as_maximum_with_model_method
     t.title = ""
     assert_predicate t, :valid?
   end
+
+  def test_validates_length_of_using_proc_as_minimum_nil_with_instance_variable
+    Topic.instance_variable_set(:@min_title_length, 5)
+    Topic.validates_length_of :title, minimum: Proc.new(&Topic.instance_variable_get(:@min_title_length))
```

Comment:
This test is not asserting what we want. You are setting an instance variable in the class, not in an instance of this class.

Reply:
Yes, as the issue says https://github.com/rails/rails/issues/40642#issuecomment-730328921 it passed as instance variable into the Proc.

Reply:
This is an instance variable of the class, not an instance variable of an instance of the class. What you need to do is  something like:

```ruby
class Topic
  def some_instance_method
  end
end

-> (topic) { topic.some_instance_method }
```

Reply:
Thanks. Updated accordingly.